source/common/upstream/upstream_impl.cc

```diff
@@ -572,6 +572,9 @@ Host::CreateConnectionData HostImplBase::createConnection(
   connection->connectionInfoSetter().enableSettingInterfaceName(
       cluster.setLocalInterfaceNameOnUpstreamConnections());
   connection->setBufferLimits(cluster.perConnectionBufferLimitBytes());
+  if (auto upstream_info = connection->streamInfo().upstreamInfo(); upstream_info) {
```

Comment:
In what cases does the streaminfo not have an upstreaminfo?

Reply:
When the socket errors out early in the code path. I'm not sure it was placed that way, but better be safe.